File: frontend/src/constants/permissions.ts
/**
 * 权限常量定义
 * 
 * 与后端 constants/permissions.go 保持一致
 */

/**
 * 项目文档权限
 */
export const ScopeProjectDocRead = 'project.doc.read';
export const ScopeProjectDocWrite = 'project.doc.write';
export const ScopeProjectAdmin = 'project.admin';

/**
 * 任务权限
 */
export const ScopeTaskRead = 'task.read';
export const ScopeTaskWrite = 'task.write';
export const ScopeTaskPlanApprove = 'task.plan.approve';

/**
 * 特性权限
 */
export const ScopeFeatureRead = 'feature.read';
export const ScopeFeatureWrite = 'feature.write';

/**
 * 会议权限
 */
export const ScopeMeetingRead = 'meeting.read';
export const ScopeMeetingWrite = 'meeting.write';

/**
 * 用户管理权限
 */
export const ScopeUserManage = 'user.manage';

/**
 * 所有可用的权限范围
 */
export const AllScopes = [
  ScopeProjectDocRead,
  ScopeProjectDocWrite,
  ScopeProjectAdmin,
  ScopeTaskRead,
  ScopeTaskWrite,
  ScopeTaskPlanApprove,
  ScopeFeatureRead,
  ScopeFeatureWrite,
  ScopeMeetingRead,
  ScopeMeetingWrite,
  ScopeUserManage,
] as const;

/**
 * 权限分组 (用于前端选择器)
 */
export interface PermissionGroup {
  title: string;
  scopes: Array<{
    label: string;
    value: string;
    description: string;
  }>;
}

export const PermissionGroups: PermissionGroup[] = [
  {
    title: '项目管理',
    scopes: [
      {
        label: '项目全局管理',
        value: ScopeProjectAdmin,
        description: '创建项目、管理项目的全局配置（系统级权限）',
      },
    ],
  },
  {
    title: '项目文档',
    scopes: [
      {
        label: '读取文档',
        value: ScopeProjectDocRead,
        description: '查看项目文档 (特性列表、架构设计等)',
      },
      {
        label: '编辑文档',
        value: ScopeProjectDocWrite,
        description: '编辑项目文档 (特性列表、架构设计等)',
      },
    ],
  },
  {
    title: '任务管理',
    scopes: [
      {
        label: '查看任务',
        value: ScopeTaskRead,
        description: '查看任务详情、需求文档、设计文档',
      },
      {
        label: '编辑任务',
        value: ScopeTaskWrite,
        description: '编辑任务详情、需求文档、设计文档',
      },
      {
        label: '审批执行计划',
        value: ScopeTaskPlanApprove,
        description: '审批或拒绝任务的执行计划',
      },
    ],
  },
  {
    title: '特性管理',
    scopes: [
      {
        label: '查看特性',
        value: ScopeFeatureRead,
        description: '查看特性列表和详情',
      },
      {
        label: '编辑特性',
        value: ScopeFeatureWrite,
        description: '创建、编辑、删除特性',
      },
    ],
  },
  {
    title: '会议管理',
    scopes: [
      {
        label: '查看会议',
        value: ScopeMeetingRead,
        description: '查看会议详情、转写记录、话题',
      },
      {
        label: '编辑会议',
        value: ScopeMeetingWrite,
        description: '编辑会议信息、话题、参会权限',
      },
    ],
  },
  {
    title: '用户管理',
    scopes: [
      {
        label: '用户管理',
        value: ScopeUserManage,
        description: '管理用户、角色、权限（系统级权限）',
      },
    ],
  },
];

/**
 * 获取权限的显示标签
 */
export function getScopeLabel(scope: string): string {
  for (const group of PermissionGroups) {
    const found = group.scopes.find(s => s.value === scope);
    if (found) return found.label;
  }
  return scope;
}

/**
 * 获取权限的描述
 */
export function getScopeDescription(scope: string): string {
  for (const group of PermissionGroups) {
    const found = group.scopes.find(s => s.value === scope);
    if (found) return found.description;
  }
  return '';
}

File: cmd/server/internal/constants/permissions.go
package constants

// Permission Scopes - 权限常量定义
const (
	// Feature Management - 特性管理
	ScopeFeatureRead  = "feature.read"
	ScopeFeatureWrite = "feature.write"

	// Task Management - 任务管理
	ScopeTaskRead  = "task.read"
	ScopeTaskWrite = "task.write"

	// Project Document - 项目文档 (新增)
	ScopeProjectDocRead  = "project.doc.read"  // 项目文档读取
	ScopeProjectDocWrite = "project.doc.write" // 项目文档编辑

	// Task Planning - 任务计划 (新增)
	ScopeTaskPlanApprove = "task.plan.approve" // 任务计划审批

	// Meeting Management - 会议管理 (新增)
	ScopeMeetingRead  = "meeting.read"  // 会议读取
	ScopeMeetingWrite = "meeting.write" // 会议编辑
)

// PermissionGroups - 权限分组配置 (供前端使用)
var PermissionGroups = map[string][]string{
	"项目文档": {
		ScopeProjectDocRead,
		ScopeProjectDocWrite,
	},
	"任务管理": {
		ScopeTaskRead,
		ScopeTaskWrite,
		ScopeTaskPlanApprove,
	},
	"会议管理": {
		ScopeMeetingRead,
		ScopeMeetingWrite,
	},
	"特性管理": {
		ScopeFeatureRead,
		ScopeFeatureWrite,
	},
}

// ValidScopes - 所有合法的 scope 集合 (用于校验)
var ValidScopes = []string{
	ScopeFeatureRead,
	ScopeFeatureWrite,
	ScopeTaskRead,
	ScopeTaskWrite,
	ScopeProjectDocRead,
	ScopeProjectDocWrite,
	ScopeTaskPlanApprove,
	ScopeMeetingRead,
	ScopeMeetingWrite,
}

// IsValidScope 检查给定的 scope 是否合法
func IsValidScope(scope string) bool {
	for _, validScope := range ValidScopes {
		if scope == validScope {
			return true
		}
	}
	return false
}

// Error Codes - 错误码定义
const (
	// Role Related Errors - 角色相关错误
	ErrCodeRoleNotFound      = "ROLE_NOT_FOUND"      // 角色不存在
	ErrCodeRoleInUse         = "ROLE_IN_USE"         // 角色正在使用中,无法删除
	ErrCodeDuplicateRoleName = "DUPLICATE_ROLE_NAME" // 角色名称重复
	ErrCodeInvalidScope      = "INVALID_SCOPE"       // 无效的权限 scope

	// User Related Errors - 用户相关错误
	ErrCodeUserNotFound = "USER_NOT_FOUND" // 用户不存在

	// Permission Related Errors - 权限相关错误
	ErrCodePermissionDenied = "PERMISSION_DENIED" // 权限不足
)

// Audit Log Actions - 审计日志动作类型
const (
	AuditActionCreateRole     = "create_role"     // 创建角色
	AuditActionUpdateRole     = "update_role"     // 更新角色
	AuditActionDeleteRole     = "delete_role"     // 删除角色
	AuditActionAssignRole     = "assign_role"     // 指派角色
	AuditActionRevokeRole     = "revoke_role"     // 撤销角色
	AuditActionChangePassword = "change_password" // 修改密码
)

// Log Prefixes - 日志前缀
const (
	LogPrefixRole     = "[ROLE]"      // 角色操作日志
	LogPrefixUserRole = "[USER_ROLE]" // 用户角色操作日志
	LogPrefixAuth     = "[AUTH]"      // 权限校验日志
	LogPrefixSecurity = "[SECURITY]"  // 安全相关日志
)
